test/test_derivatives.py

```diff
@@ -97,7 +97,7 @@ def test_scale_factor():
     topography2 = UniformLineScan(topography.heights()[1::2], sx, periodic=True)
 
     d1 = topography.derivative(1, scale_factor=1)
-    d2 = topography.derivative(1, scale_factor=2)
+    d2 = topography.derivative(1, scale_factor=np.uint32(2))
```

Comment:
Can the other type of scale factors also be tested here?
From the docstring: "int or list of ints or list of tuples of ints"

Reply:
I added this test